proto/pdpb.proto

```diff
@@ -1,303 +1,321 @@
-    optional int64 logical  = 2 [(gogoproto.nullable) = false];
+    int64 physical = 1;
+    int64 logical = 2;
 }
 
 message TsoResponse {
-    optional uint32 count        = 1 [(gogoproto.nullable) = false];
-    optional Timestamp timestamp = 2 [(gogoproto.nullable) = false];
+    ResponseHeader header = 1;
+
+    uint32 count = 2;
+    Timestamp timestamp = 3;
 }
 
 message BootstrapRequest {
-    optional metapb.Store store   = 1;
-    optional metapb.Region region = 2;
+    RequestHeader header = 1;
+
+    metapb.Store store = 2;
+    metapb.Region region = 3;
 }
 
 message BootstrapResponse {
-    
+    ResponseHeader header = 1;
 }
 
 message IsBootstrappedRequest {
-    
+    RequestHeader header = 1;
 }
 
 message IsBootstrappedResponse {
-    optional bool bootstrapped = 1;
-}
+    ResponseHeader header = 1;
 
-message AllocIdRequest {
+    bool bootstrapped = 2;
+}
 
+message AllocIDRequest {
+    RequestHeader header = 1;
 }
 
-message AllocIdResponse {
-    optional uint64 id             = 1 [(gogoproto.nullable) = false];
+message AllocIDResponse {
+    ResponseHeader header = 1;
+
+    uint64 id = 2;
 }
 
 message GetStoreRequest {
-    optional uint64 store_id       = 1 [(gogoproto.nullable) = false];
+    RequestHeader header = 1;
+
+    uint64 store_id = 2;
 }
 
 message GetStoreResponse {
-    optional metapb.Store store     = 1;
+    ResponseHeader header = 1;
+
+    metapb.Store store = 2;
+}
+
+message PutStoreRequest {
+    RequestHeader header = 1;
+
+    metapb.Store store = 2;
+}
+
+message PutStoreResponse {
+    ResponseHeader header = 1;
 }
 
 message GetRegionRequest {
-    optional bytes region_key      = 1;
+    RequestHeader header = 1;
+
+    bytes region_key = 2;
 }
 
 message GetRegionResponse {
-    optional metapb.Region region   = 1;
-    optional metapb.Peer leader     = 2;
+    ResponseHeader header = 1;
+
+    metapb.Region region = 2;
+    metapb.Peer leader = 3;
 }
 
 message GetRegionByIDRequest {
-    optional uint64 region_id      = 1 [(gogoproto.nullable) = false];
+    RequestHeader header = 1;
+
+    uint64 region_id = 2;
 }
 
 // Use GetRegionResponse as the response of GetRegionByIDRequest.
 
 message GetClusterConfigRequest {
-
+    RequestHeader header = 1;
 }
 
 message GetClusterConfigResponse {
-    optional metapb.Cluster cluster = 1;
+    ResponseHeader header = 1;
+
+    metapb.Cluster cluster = 2;
 }
 
+message PutClusterConfigRequest {
+    RequestHeader header = 1;
 
-message PutStoreRequest {
-    optional metapb.Store store     = 1;
+    metapb.Cluster cluster = 2;
 }
 
-message PutStoreResponse {
+message PutClusterConfigResponse {
+    ResponseHeader header = 1;
 }
 
-message PDMember {
-    optional string name        = 1;
-    repeated string client_urls = 2;
-    repeated string peer_urls   = 3;
+message Member {
+    // name is the name of the PD member.
+    string name = 1;
+    // member_id is the unique id of the PD member.
+    uint64 member_id = 2;
```

Comment:
So that we can easily identify those members. BTW this message is a clone of etcd's member. https://github.com/coreos/etcd/blob/43740a8d3c36f37f42754074856b7fc1c76beab5/etcdserver/etcdserverpb/rpc.proto#L717